monai/apps/manifest/utils.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+from monai.apps.utils import download_url
+from monai.config import PathLike
+from monai.utils import optional_import
+
+validate, _ = optional_import("jsonschema", name="validate")
+ValidationError, _ = optional_import("jsonschema.exceptions", name="ValidationError")
+
+
+def verify_metadata(
+    metadata: Union[Dict, str],
+    schema_url: str,
+    filepath: PathLike,
+    result_path: Optional[PathLike] = None,
+    create_dir: bool = True,
+    hash_val: Optional[str] = None,
+    **kwargs,
+):
+    """
+    Verify the provided metadata dictonary or file based on the predefined schema.
+
+    Args:
+        metadata: source meta data to verify.
+        schema_url: URL to download the expected schema file.
+        filepath: file path to store the downloaded schema.
+        result_path: if not None, save the validation error into result file.
+        create_dir: whether to create directories if not existing.
+        hash_val: if not None, define the hash value to verify the downloaded schema file.
+        kwargs: other arguments for `jsonschema.validate()`. for more details:
+            https://python-jsonschema.readthedocs.io/en/stable/validate/#jsonschema.validate.
+
+    """
+
+    def _check_dir(path: Path):
+        path_dir = path.parent
+        if not path_dir.exists():
+            if create_dir:
+                path_dir.mkdir(parents=True)
+            else:
+                raise ValueError(f"the directory of specified path is not existing: {path_dir}.")
+
+    filepath = Path(filepath)
+    _check_dir(path=filepath)
+    download_url(url=schema_url, filepath=filepath, hash_val=hash_val, hash_type="md5", progress=True)
```

Comment:
Instead of using `download_url` we could just use `urllib.request.urlopen` and not need the temp directory.

Reply:
Hi @ericspod ,

I think `download_url` also provides the ability to optionally verify the hash value, which may be necessary to ensure the downloaded `schema` file is correct.
What do you think?

Thanks.

Reply:
I think it can also avoid repeatedly downloading the same schema file, especially when the deployed environment has limited network access.

Thanks.